test/trajectory_generation_test.cpp

```diff
@@ -1128,6 +1128,163 @@ TEST_F(TrajectoryGenerationTest, SingleJointOscillation)
   EXPECT_NEAR(output_trajectories_[0].elapsed_times[1] - output_trajectories_[0].elapsed_times[0], timestep_,
               timestep_tolerance);
 }
+
+TEST_F(TrajectoryGenerationTest, VelocityLimit)
+{
+  // Velocity limit is hit
+
+  const double max_duration = 20;
+
+  KinematicState joint_state;
+
+  current_joint_states_[0] = joint_state;
+  current_joint_states_[1] = joint_state;
+  current_joint_states_[2] = joint_state;
+
+  joint_state.position = -0.995;
+  joint_state.velocity = 0;
+  joint_state.acceleration = 0;
+  goal_joint_states_[0] = joint_state;
+  goal_joint_states_[1] = joint_state;
+  goal_joint_states_[2] = joint_state;
+
+  // Unlimited max velocity: 1.86513, Max accel: 5.74272, Max jerk: 54.4166
```

Comment:
Nitpick. Just because this sentence confused me at first.

```suggestion
  // Maximums prior to trackjoint: velocity: 1.86513, accel: 5.74272, jerk: 54.4166
```